File: fibo.c
#include <stdio.h>
#include <time.h>

// フィボナッチ数列
long fibo(int n)
{
    if (n <= 0)
    {
        return 0;
    }
    if (n == 1)
    {
        return 1;
    }
    return fibo(n - 1) + fibo(n - 2);
}

int main(void)
{
    int n = 39;
    time_t t0, t1;
    time(&t0);
    printf("fibo(%d) = %ld\n", n, fibo(n));
    time(&t1);
    printf("%ld 秒\n", t1 - t0);
    return 0;
}

File: fibo.py
#!/usr/bin/python3
import time
import sys
import argparse

parser = argparse.ArgumentParser(description="フィボナッチ数列の計算時間を計測")
parser.add_argument("integers", metavar='N', type=int, help="フィボナッチ数列の入力")

args = parser.parse_args()

# フィボナッチ数列
def fibo(n):
    if n <= 0:
        return 0
    if n == 1:
        return 1
    return fibo(n - 1) + fibo(n - 2)

n = args.integers
print(sys.version)
t0 = time.time()
print("fibo(%d) = %d" % (n, fibo(n)))
print(time.time() - t0, "秒")
